mantidproject/mantid
# Support loading 1D plots from MantidPlot project files

**Description of work.**
This PR adds the ability to read 1D plots from MantidPlot project files into workbench. Currently, this allows the following to be reproduced:

- Single 1D plots
- 1D plots with errors
- Log scaled plots
- Multiple 1d plots
- Tiled plots

Matplotlib settings which cannot be inferred from the MantidPlot project file have been given default values.

**To test:**
- Open mantidplot
- Load some workspaces
- Make different types of 1d plots
- Save the project using File->Save project as
- Open workbench and load that project
- The workspaces and 1d plots should be reproduced



Fixes #30043 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
